plugin/src/main/java/org/opensearch/ml/sdkclient/DDBOpenSearchClient.java

```diff
@@ -155,16 +200,140 @@ public CompletionStage<DeleteDataObjectResponse> deleteDataObjectAsync(DeleteDat
         }), executor);
     }
 
+    /**
+     * DDB data needs to be synced with opensearch cluster. {@link RemoteClusterIndicesClient} will then be used to
+     * search data in opensearch cluster.
```

Comment:
For my education, what's happening here?  

Are we writing to DDB and reading from OS for all "fetch" operations or are we only syncing up for search?

Reply:
DDBOpenSearchClient will make use of DDB for CRUD operations. Data written to DDB will be synced with OpenSearch using offline integration between DDB and opensearch. SDKClient will then rely on opensearch only for search operations.